android/KMAPro/kMAPro/src/main/java/com/tavultesoft/kmapro/PackageActivity.java

```diff
@@ -0,0 +1,260 @@
+      String message = "Invalid package version in " + kmpFile.getName();
+      showErrorDialog(context, pkgId, message);
+    }
+
+    try {
+      tempPackagePath = PackageProcessor.unzipKMP(kmpFile);
+    } catch (Exception e) {
+      String message = "Failed to extract\n" + kmpFile.getAbsolutePath();
+      showErrorDialog(context, pkgId, message);
+    }
+
+    final ActionBar actionBar = getActionBar();
+    actionBar.setLogo(R.drawable.keyman_logo);
+    actionBar.setDisplayShowHomeEnabled(false);
+    actionBar.setDisplayShowTitleEnabled(false);
+    actionBar.setDisplayShowCustomEnabled(true);
+    actionBar.setBackgroundDrawable(MainActivity.getActionBarDrawable(this));
+    TextView packageActivityTitle = new TextView(this);
+    packageActivityTitle.setWidth((int) getResources().getDimension(R.dimen.package_label_width));
+    packageActivityTitle.setTextSize(getResources().getDimension(R.dimen.package_label_textsize));
+    packageActivityTitle.setGravity(Gravity.CENTER);
+
+    String titleStr = "Install Keyboard Package";
+    if (version != null) {
+      titleStr += " " + version;
+    }
+    packageActivityTitle.setText(titleStr);
+    actionBar.setCustomView(packageActivityTitle);
+
+    setContentView(R.layout.activity_package_installer);
+
+    final Button installButton = (Button) findViewById(R.id.installButton);
+    final Button cancelButton = (Button) findViewById(R.id.cancelButton);
+
+    webView = (WebView) findViewById(R.id.webView);
+    webView.getSettings().setLayoutAlgorithm(WebSettings.LayoutAlgorithm.NORMAL);
+    webView.getSettings().setJavaScriptEnabled(true);
+    webView.getSettings().setUseWideViewPort(true);
+    webView.getSettings().setLoadWithOverviewMode(true);
+    webView.getSettings().setBuiltInZoomControls(true);
+    webView.getSettings().setSupportZoom(true);
+    webView.getSettings().setTextZoom(200);
+    webView.setVerticalScrollBarEnabled(true);
+    webView.setHorizontalScrollBarEnabled(true);
+    webView.setLayerType(View.LAYER_TYPE_SOFTWARE, null);
+
+    webView.setWebChromeClient(new WebChromeClient() {
+    });
+    webView.setWebViewClient(new WebViewClient() {
+      @Override
+      public void onReceivedError(WebView view, int errorCode, String description, String failingUrl) {
+      }
+
+      @Override
+      public boolean shouldOverrideUrlLoading(WebView view, String url) {
+        if (!url.toLowerCase().equals("about:blank"))
+          view.loadUrl(url);
+
+        return true;
+      }
+
+      @Override
+      public void onPageStarted(WebView view, String url, Bitmap favicon) {
+      }
+
+      @Override
+      public void onPageFinished(WebView view, String url) {
+      }
+    });
+
+    // Determine if KMP contains welcome.htm (case-insensitive) to display
+    FileFilter welcomeFilter = new FileFilter() {
+      @Override
+      public boolean accept(File pathname) {
+        if (pathname.isFile() && pathname.getName().equalsIgnoreCase("welcome.htm")) {
```

Comment:
Maybe check for "welcome.html" as well?

Reply:
per @mcdurdin due to Windows 9x, it will always be "welcome.htm"